### PR TITLE
feat: add action menu extension

### PR DESCRIPTION
**Pull request by ashutosh16**
Ref: https://github.com/argoproj/argo-cd/pull/19620

Screenshot:
Screenshot 2024-08-20 at 22 24 00

### CODE REVIEW

**Review by leoluz — CHANGES_REQUESTED** [review]
Please update the UI extensions documentation explaining the new type

**Review by rbreeze — APPROVED** [review]
LGTM once documentation is added

File: stories/top-bar.stories.tsx
Comment (agilgur5) [inline stories/top-bar.stories.tsx:6]: So I do have a pending PR to update all this old syntax: #568 . I'd prefer not to have more stories before that is merged

**Review by agilgur5 — COMMENTED** [review]
What is this extension adding? In the screenshot is it the new button? That wouldn't need a change in `argo-ui`.

I would also strongly advise against CD UI extensions reaching into `argo-ui` given the deprecated state of this repo per pinned issue #453 . Giving a deprecated repo a new feature that user-facing extensions hook into sounds like exposing far too much, especially for a not very reliable repo.
 I would also advise against that as this repo is not supposed to be specific to CD.

File: src/components/top-bar/top-bar.tsx
Comment (agilgur5) [inline src/components/top-bar/top-bar.tsx:97]: Adding another button does not require an `argo-ui` change, you can just add another `item` to the list in CD.

**Review by agilgur5 — CHANGES_REQUESTED** [review]

File: stories/top-bar.stories.tsx
Comment (ashutosh16) [reply stories/top-bar.stories.tsx]: agree, i have remove the stories and will add when the pending  PRs get merged

File: src/components/top-bar/top-bar.tsx
Comment (ashutosh16) [reply src/components/top-bar/top-bar.tsx:97]: I still want to pursue these changes, and I think this is an easy solution that does not require any extensions or argo. If we have a better solution in the future, I will be happy to review it.

Comment (agilgur5) [reply src/components/top-bar/top-bar.tsx:97]: > I think this is an easy solution that does not require any extensions or argo

It's definitely not "easy" given how invasive it is, and it adds extensions to `argo-ui`, which does in fact add logic to Argo

Comment (agilgur5) [reply src/components/top-bar/top-bar.tsx:97]: See the concrete guidance I left downstream for how to avoid a change in `argo-ui`: https://github.com/argoproj/argo-cd/pull/19620/files/3e67e6dbda7c7f716a35164c636ad174df0e7396#r1727992070

**Comment by agilgur5** [comment]
I think this can be closed given the latest changes in https://github.com/argoproj/argo-cd/pull/19620 that obviate the need for this per [my suggestions](https://github.com/argoproj/argo-ui/pull/576#discussion_r1727997681).

CD folks should probably double-check my review there since I'm not a regular contributor there (nor have approver permissions there)

**Pull request closed without merging**
